app/src/main/java/com/example/atheneum/activities/ViewProfileActivity.java

```diff
@@ -0,0 +1,117 @@
+import com.example.atheneum.models.User;
+import com.example.atheneum.utils.FirebaseAuthUtils;
+import com.example.atheneum.utils.PhotoUtils;
+import com.example.atheneum.viewmodels.UserViewModel;
+import com.example.atheneum.viewmodels.UserViewModelFactory;
+import com.google.firebase.auth.FirebaseUser;
+
+import java.util.ArrayList;
+
+/**
+ * The activity for users to view their own profiles
+ * and other profiles
+ */
+public class ViewProfileActivity extends AppCompatActivity {
+    private static final String TAG = "View Profile Activity";
+
+    /**
+     * Override onCreate method to produce layout for individual user profiles
+     * @param savedInstanceState Bundle environment data
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_view_profile);
+
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        setActionBarTitle(getResources().getString(R.string.title_activity_view_profile));
+
+        Intent intent = getIntent();
+        User user = (User) intent.getSerializableExtra("user");
+
+        final ImageView profilePicture = findViewById(R.id.user_profile_pic);
+//        try {
+//            String userPic = user.getPhotos().get(0);
+//            Bitmap bitmapPhoto = StringToBitMap(userPic);
+//            profilePicture.setImageBitmap(bitmapPhoto);
+//        } catch (Exception ignore) {
+//
+//        }
+
+        final TextView username = findViewById(R.id.username);
+        final TextView phone = findViewById(R.id.phone);
+        final TextView borrower_rating = findViewById(R.id.borrower);
+        final TextView owner_rating = findViewById(R.id.owner);
+
+        UserViewModelFactory factory = new UserViewModelFactory(user.getUserID());
+        UserViewModel userViewModel = ViewModelProviders.of(this, factory).get(UserViewModel.class);
+        LiveData<User> userLiveData = userViewModel.getUserLiveData();
+        userLiveData.observe(this, new Observer<User>() {
+            @Override
+            public void onChanged(@Nullable User user) {
+                if (user != null) {
+                    username.setText(user.getUserName());
+                    phone.setText(user.getPhoneNumber());
+                    borrower_rating.setText(Double.toString(user.getBorrowerRate()));
+                    owner_rating.setText(Double.toString(user.getOwnerRate()));
+                    ArrayList<String> photos = user.getPhotos();
+                    if (!photos.isEmpty()) {
+                        String userPic = user.getPhotos().get(0);
+                        Bitmap bitmapPhoto = PhotoUtils.DecodeBase64BitmapPhoto(userPic);
+                        profilePicture.setImageBitmap(bitmapPhoto);
+                    }
+
+                }
+            }
+        });
+
+        FirebaseUser currUser = FirebaseAuthUtils.getCurrentUser();
```

Comment:
You should wrap this section of code in a check for FirebaseAuthUtils.isAuthenticated() since currUser can be null if we lose the authentication. If we lose authentication and we try to call currUser.getUid() you will hit a null pointer exception.

Reply:
Looks like this is fixed